arrow-libs/core/arrow-atomic/build.gradle.kts

```
@@ -46,10 +46,8 @@ kotlin {
   }
 }
 
-tasks.withType<KotlinCompile>().configureEach {
-  kotlinOptions {
-    freeCompilerArgs = freeCompilerArgs + "-Xexpect-actual-classes"
-  }
+tasks.withType<KotlinCompile> {
+  kotlinOptions.freeCompilerArgs += "-Xexpect-actual-classes"
 }
 
```

I highly recommend to keep the `configureEach`

```suggestion
tasks.withType<KotlinCompile>.configureEach {
  kotlinOptions.freeCompilerArgs += "-Xexpect-actual-classes"
}
```

---

Any particular reason for this? (Gradle is all magic to me)

---

Yep, `configureEach` is lazy, the other isn't. If I remember correctly, it is calling `all` under the hood.

I hope they mark all of those APIs as deprecated in Gradle 9.